src/docs/fund.yaml

```diff
@@ -0,0 +1,376 @@
+          description: user not sigend in
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: boolean
+                    example: true
+                  message:
+                    type: string
+                    example: "Invalid Token: No authorization token was found"
+        422:
+          description: This status is returned when the request is missing required fields or has an invalid values.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ValidationError"
+        500:
+          description: Internal server error
+  /api/fund/{id}/donate:
+    post:
+      tags:
+        - Fund
+      summary: Donate to a fund
+      description: Donates to a fund
+      operationId: donate
+      produces:
+        - application/json
+      consumes:
+        - application/json
+      parameters:
+        - in: path
+          name: id
+          description: The fund id
+          required: true
+          type: string
+        - in: body
+          name: body
+          required: true
+          schema:
+            properties:
+              amount:
+                type: number
+                description: The amount to donate
+                format: double
+                example: 10.00
+      responses:
+        201:
+          description: Successfully donated to fund
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Donation successful
+        422:
+          description: This status is returned when the request is missing required fields or has an invalid values.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ValidationError"
+        500:
+          description: Internal server error
+  /api/funds/:
+    get:
+      tags:
+        - Fund
+      description: "funds route"
+      parameters:
+        - in: query
+          name: publisherId
+          type: integer
+          collectionFormat: multi
+        - name: categories
+          in: query
+          schema:
+            type: array
+            items:
+              type: string
+              example: "category1"
+              collectionFormat: multi
+        - name: lastDate
+          in: query
+          schema:
+            type: string
+            format: date
+            example: 2022-01-01
+        - name: currentDate
+          in: query
+          schema:
+            type: string
+            format: date
+            example: 2022-01-05
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schema/Fund"
```

Comment:
`schemas`

In fact, you should not use Fund directly because in it the publisherId is populated but you are not populating it in your endpoint.